runtime/common/src/apis.rs

```diff
@@ -262,18 +262,53 @@ macro_rules! impl_runtime_apis_plus_common {
 					};
 					let is_transactional = false;
 					let validate = true;
+
+					let estimated_transaction_len = (data.encode().len() +
+						max_fee_per_gas.encode().len() +
+						max_priority_fee_per_gas.encode().len() +
```

Comment:
Please avoid dynamic encoding, just add 32 if is_some

Reply:
the goal is to optimize I agree but let not forget this is never onchain calculation, only on-demand for rpc nodes. Lets consider further changes negligible.